src/System.Net.Http.WinHttpHandler/tests/System.Net.Http.WinHttpHandler.Tests.csproj

```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
@mmitche I completely agree with @davidsh. My definition for "Unit Test" is the one mentioned in [The Art of Unit Testing](http://www.amazon.com/The-Art-Unit-Testing-Examples/dp/1933988274). 

Large changes to the code are dangerous without having enough tests covering the expected behaviors. Behavior coverage of stateful and timing-sensitive APIs such as networking stacks is only achievable by simulating the states within the same process instead of tying the tests to complicated infrastructure that would allow packet capture and modification. 

Given the large number of permutations, it would also be impracticable to test all combinations using a real network: think of various platform version support (Win7 to Win10), network failures, network latency, proxy types, authentication, encryption, certificates, etc.

As @davidsh pointed out, we are also thinking of ways to expose Functional tests which will require a real server configured. Functional tests will be much slower, harder to setup (every developer will be required to have a local server) and will mostly validate the happy-paths.